bank-account-application/src/main/java/com/banking/ibi/config/JwtTokenValidator.java

```diff
@@ -36,18 +37,28 @@ public JwtTokenValidator(JwtTokenProvider jwtTokenProvider, UserServices userSer
 		this.userRepository = userRepository;
 
 	}
+	
+	private Cookie getAuthCookie(HttpServletRequest request) {
+		if(request==null || request.getCookies().length==0 || request.getCookies()[0].getValue()== null) return null;
+		Cookie[] cookies = request.getCookies();
+		for(Cookie cookie : cookies) {
+			if(cookie.getName().equals("auth_token")) return cookie;
+		}
+		return null;
+	}
 
 	@Override
 	protected void doFilterInternal(@NonNull HttpServletRequest request, @NonNull HttpServletResponse response,
 			@NonNull FilterChain filterChain) throws ServletException, IOException {
 		
-
-		if (request.getCookies() == null|| request.getCookies().length==0||request.getCookies()[0].getValue()== null) {
+		Cookie authCookie = getAuthCookie(request);
```

Comment:
We were just using first cookie and expecting it to be correct one. Thats not a guarantee so implemented a new method for that